python/tskit/tables.py

```diff
@@ -2924,20 +2945,44 @@ def drop_index(self):
         """
         self._ll_tables.drop_index()
 
-    def subset(self, nodes, record_provenance=True):
+    def subset(
+        self,
+        nodes,
+        record_provenance=True,
+        filter_populations=True,
+        filter_individuals=True,
+        filter_sites=True,
+        canonicalise=False,
+    ):
         """
         Modifies the tables in place to contain only the entries referring to
         the provided list of nodes, with nodes reordered according to the order
         they appear in the list. See :meth:`TreeSequence.subset` for a more
         detailed description.
 
+        Note: the tables can be completely unsorted.
+
         :param list nodes: The list of nodes for which to retain information. This
             may be a numpy array (or array-like) object (dtype=np.int32).
         :param bool record_provenance: Whether to record a provenance entry
             in the provenance table for this operation.
+        :param bool filter_populations: Whether to remove populations not referenced by
+            retained nodes. If False, the population table will remain unchanged.
+        :param bool filter_individuals: Whether to remove individuals not referenced by
+            retained nodes. If False, the individuals table will remain unchanged.
+        :param bool filter_sites: Whether to remove sites not referenced by
+            retained mutations. If False, the site table will remain unchanged.
+        :param bool canonicalise: If True, retains all unused entries, putting
```

Comment:
I don't think the canonical form should include any redundant information (see note above), so I don't think this is the right name for the parameter. Seems like we're getting into a bit of a mess here just to keep unreferenced objects in the data model. Is this something that we really need?